cartesian_trajectory_interpolation/src/cartesian_trajectory.cpp

```diff
@@ -55,8 +56,31 @@ bool CartesianTrajectory::init(const cartesian_control_msgs::CartesianTrajectory
     {
       return false;
     }
-    CartesianTrajectorySegment s(i->time_from_start.toSec(), CartesianState(*i), std::next(i)->time_from_start.toSec(),
-                                 CartesianState(*std::next(i)));
+
+    CartesianState state = CartesianState(*i);
+    CartesianState next_state = CartesianState(*std::next(i));
+
+    if (sign_flipped)
+    {
+      state.q.x() = state.q.x() * -1;
+      state.q.y() = state.q.y() * -1;
+      state.q.z() = state.q.z() * -1;
+      state.q.w() = state.q.w() * -1;
+      sign_flipped = false;
+    }
+
+    // If the dot product is negative, we change the sing of the orientation to not travel the long way around
+    double dot_product = state.q.x() * next_state.q.x() + state.q.y() * next_state.q.y() +
+                         state.q.z() * next_state.q.z() + state.q.w() * next_state.q.w();
```

Comment:
```suggestion
    double dot_product = state.q.dot(next_state.q);
```

Reply:
Less verbose and probably performance optimized. Do you want to keep the explicit computation for the sake of clarity?